File: graph/WeightedUndirectedGraph.cs
using System;
using System.Collections.Generic;

class Pair : IComparable<Pair>
{
    public int First, Second;

    public Pair(int first, int second)
    {
        First = first;
        Second = second;
    }

    public int CompareTo(Pair other)
    {
        if (First != other.First) return First - other.First;
        return Second - other.Second;
    }

    public override string ToString() => First + " " + Second;
}

class Triplet : IComparable<Triplet>
{
    public int First, Second, Third;

    public Triplet(int first, int second, int third)
    {
        First = first;
        Second = second;
        Third = third;
    }

    public int CompareTo(Triplet other)
    {
        if (First != other.First) return First - other.First;
        if (Second != other.Second) return Second - other.Second;
        return Third - other.Third;
    }

    public override string ToString() => First + " " + Second + " " + Third;
}

class WeightedUndirectedGraph
{
    public List<List<Pair>> AdjList;
    public List<Triplet> EdgeList;

    public WeightedUndirectedGraph(int n)
    {
        AdjList = new List<List<Pair>>();
        EdgeList = new List<Triplet>();
        for (int i = 0; i <= n; i++)
            AdjList.Add(new List<Pair>());
    }

    public void AddEdge(int u, int v, int w)
    {
        EdgeList.Add(new Triplet(u, v, w));
        AdjList[u].Add(new Pair(v, w));
        AdjList[v].Add(new Pair(u, w));
    }

    public void PrintEdgeList()
    {
        Console.WriteLine("Edge List:");
        Console.WriteLine(string.Join("\n", EdgeList));
    }

    public void PrintAdjList()
    {
        Console.WriteLine("Adjacency List:");
        for (int i = 0; i < AdjList.Count; i++)
        {
            if (AdjList[i].Count == 0)
                Console.WriteLine(i + ": No neighbors");
            else
            {
                string neighbors = string.Join(" ", AdjList[i].ConvertAll(p => "(" + p + ")"));
                Console.WriteLine(i + ": " + neighbors);
            }
        }
    }
}

class Program
{
    static void Main()
    {
        string[] input = Console.In.ReadToEnd().Split(
            new char[] { ' ', '\n', '\r', '\t' },
            StringSplitOptions.RemoveEmptyEntries
        );

        int idx = 0;
        Func<int> ReadInt = () => int.Parse(input[idx++]);  // ← Fix: Func<int> instead of local function

        int t = ReadInt();

        while (t-- > 0)
        {
            int V = ReadInt();
            int E = ReadInt();

            WeightedUndirectedGraph g = new WeightedUndirectedGraph(V);

            for (int i = 0; i < E; i++)
            {
                int u = ReadInt();
                int v = ReadInt();
                int w = ReadInt();
                g.AddEdge(u, v, w);
            }

            // SOLUTION LOGIC HERE

            g.PrintEdgeList();
            g.PrintAdjList();
        }
    }
}

File: graph/WeightedUndirectedGraph.cpp
#include <bits/stdc++.h>
using namespace std;

struct Pair {
    int first, second;

    Pair(int f, int s) : first(f), second(s) {}

    bool operator<(const Pair& o) const {
        if (first != o.first) return first < o.first;
        return second < o.second;
    }

    bool operator==(const Pair& o) const {
        return first == o.first && second == o.second;
    }

    friend ostream& operator<<(ostream& os, const Pair& p) {
        return os << p.first << " " << p.second;
    }
};

struct Triplet {
    int first, second, third;

    Triplet(int f, int s, int t) : first(f), second(s), third(t) {}

    bool operator<(const Triplet& o) const {
        if (first != o.first) return first < o.first;
        if (second != o.second) return second < o.second;
        return third < o.third;
    }

    bool operator==(const Triplet& o) const {
        return first == o.first && second == o.second && third == o.third;
    }

    friend ostream& operator<<(ostream& os, const Triplet& t) {
        return os << t.first << " " << t.second << " " << t.third;
    }
};

class WeightedUndirectedGraph {
public:
    vector<vector<Pair>> adjList;
    vector<Triplet> edgeList;

    WeightedUndirectedGraph(int n) {
        adjList.resize(n + 1);
    }

    void addEdge(int u, int v, int w) {
        edgeList.push_back(Triplet(u, v, w));
        adjList[u].push_back(Pair(v, w));
        adjList[v].push_back(Pair(u, w));
    }

    void printEdgeList() {
        cout << "Edge List:\n";
        for (int i = 0; i < (int)edgeList.size(); i++) {
            if (i > 0) cout << "\n";
            cout << edgeList[i];
        }
        cout << "\n";
    }

    void printAdjList() {
        cout << "Adjacency List:\n";
        for (int i = 0; i < (int)adjList.size(); i++) {
            cout << i << ": ";
            if (adjList[i].empty()) {
                cout << "No neighbors";
            } else {
                for (int j = 0; j < (int)adjList[i].size(); j++) {
                    if (j > 0) cout << " ";
                    cout << "(" << adjList[i][j] << ")";
                }
            }
            cout << "\n";
        }
    }
};

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int t;
    cin >> t;

    while (t--) {
        int V, E;
        cin >> V >> E;

        WeightedUndirectedGraph g(V);

        for (int i = 0; i < E; i++) {
            int u, v, w;
            cin >> u >> v >> w;
            g.addEdge(u, v, w);
        }

        // SOLUTION LOGIC HERE

        g.printEdgeList();
        g.printAdjList();
    }

    return 0;
}
